### PR TITLE
Update the CreateModel API for current Juju 2.

### PR DESCRIPTION
**Pull request by frankban**
Also update the legacy API for compatibility.

### CODE REVIEW

**Comment by frankban** [comment]
QA: in gijoe, create new models.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5766/
Test PASSed.

File: jujugui/static/gui/src/app/components/user-profile/create-model-button/test-create-model-button.js [hunk 1]
```diff
@@ -32,19 +32,19 @@ describe('CreateModelButton', () => {
     controllerAPI = {
       findFacadeVersion: sinon.stub(),
       get: sinon.stub().returns('default'),
-      createModel: (modelName, userName, callback) => {
+      createModel: (modelName, userTag, args, callback) => {
         assert.equal(modelName, 'newmodelname', 'model name not set properly');
-        assert.equal(userName, 'test-owner', 'user name not set properly');
+        assert.equal(userTag, 'user-dalek', 'user tag not set properly');
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/user-profile/create-model-button/test-create-model-button.js]: Ha ha, OK, I admit, 'user-dalek' is much more fun than 'test-owner'. Good change :wink:

File: jujugui/static/gui/src/app/store/env/controller-api.js [hunk 2]
```diff
@@ -504,40 +504,66 @@ YUI.add('juju-controller-api', function(Y) {
...
+        - region: the name of the cloud region to create the model in. If the
+          cloud does not support regions, this must be empty/undefined. If this
+          is empty/undefined, and cloudTag is empty/undefined, the model will
+          be created in the same region as the controller model;
+        - credentialTag: the tag of the cloud credential to use for managing
+          the model's resources. If the cloud does not require credentials
+          this may be empty/undefined. If this is empty/undefined and the
+          owner is the controller owner then the same credential used for the
+          controller model will be used. A credential tag is a credential name
+          prefixed with "cloudcred-".
```
Comment (kadams54) [inline jujugui/static/gui/src/app/store/env/controller-api.js]: You win the Documentation Cup: :trophy:

File: jujugui/static/gui/src/app/store/env/legacy-api.js [hunk 3]
```diff
@@ -2399,35 +2399,42 @@ YUI.add('juju-env-legacy-api', function(Y) {
...
-        intermediateCallback = this._handleCreateModel.bind(null, callback);
-      } else {
-        intermediateCallback = function(callback, data) {
-          console.log('createModel done: err:', data.Error);
-        };
-      }
-      var facade = 'EnvironmentManager';
-      var request = 'CreateEnvironment';
+    createModel: function(name, userTag, args, callback) {
+      const handler = function(userCallback, data) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/store/env/legacy-api.js]: You don't need to use bind for this you can just use the outer callback. 

```JavaScript
const handler = data => {
  if (!callback) {
    ...
  }
...
```

**Comment by hatched** [comment]
Thanks @frankban :+1: with a trivial request. I'll qa shortly

**Comment by kadams54** [comment]
QA is :ok: and code is :+1: from me - thanks @frankban!

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**